openmc/plots.py

```diff
@@ -341,78 +536,70 @@ def colorize(self, geometry, seed=1):
         cv.check_greater_than('seed', seed, 1, equality=True)
 
         # Get collections of the domains which will be plotted
-        if self.color is 'mat':
-            domains = geometry.get_all_materials()
+        if self.color_by == 'material':
+            domains = geometry.get_all_materials().values()
         else:
-            domains = geometry.get_all_cells()
+            domains = geometry.get_all_cells().values()
 
         # Set the seed for the random number generator
         np.random.seed(seed)
 
         # Generate random colors for each feature
-        self.col_spec = {}
-        for domain_id in domains:
-            r = np.random.randint(0, 256)
-            g = np.random.randint(0, 256)
-            b = np.random.randint(0, 256)
-            self.col_spec[domain_id] = (r, g, b)
+        for domain in domains:
+            self.colors[domain] = np.random.randint(0, 256, (3,))
 
     def highlight_domains(self, geometry, domains, seed=1,
                           alpha=0.5, background='gray'):
         """Use alpha compositing to highlight one or more domains in the plot.
 
-        This routine generates a color scheme and applies alpha compositing
-        to make all domains except the highlighted ones appear partially
+        This routine generates a color scheme and applies alpha compositing to
+        make all domains except the highlighted ones appear partially
         transparent.
 
         Parameters
         ----------
         geometry : openmc.Geometry
             The geometry for which the plot is defined
-        domains : Iterable of Integral
+        domains : Iterable of openmc.Cell or openmc.Material
             A collection of the domain IDs to highlight in the plot
-        seed : Integral
+        seed : int
             The random number seed used to generate the color scheme
-        alpha : Real in [0,1]
+        alpha : float
```

Comment:
Yes, but the purpose of the qualifier is to indicate the type. The restriction should be noted in the description (I'll add that in).